packages/react/src/use-assistant.ts

```diff
@@ -160,67 +169,50 @@ export function useAssistant({
         throw new Error('The response body is empty.');
       }
 
-      for await (const { type, value } of readDataStream(
-        result.body.getReader(),
-      )) {
-        switch (type) {
-          case 'assistant_message': {
-            setMessages(messages => [
-              ...messages,
-              {
-                id: value.id,
-                role: value.role,
-                content: value.content[0].text.value,
-              },
-            ]);
-            break;
-          }
-
-          case 'text': {
-            // text delta - add to last message:
-            setMessages(messages => {
-              const lastMessage = messages[messages.length - 1];
-              return [
-                ...messages.slice(0, messages.length - 1),
-                {
-                  id: lastMessage.id,
-                  role: lastMessage.role,
-                  content: lastMessage.content + value,
-                },
-              ];
-            });
+      for await (const { value } of readDataStream(result.body.getReader())) {
+        const { event, data } = value as AssistantStreamPart;
+        setStatus('in_progress');
+        setThreadStatus(event);
 
+        switch (event) {
+          case 'thread.run.created': {
+            setThreadId(data.threadId);
             break;
           }
 
-          case 'data_message': {
+          case 'thread.message.created': {
             setMessages(messages => [
               ...messages,
               {
-                id: value.id ?? generateId(),
-                role: 'data',
+                id: data.id,
+                role: 'assistant',
                 content: '',
-                data: value.data,
```

Comment:
do we still have the ability to send data? i think it was for custom data

Reply:
I don't think `useAssistant` supported appending custom data, correct me if I'm wrong though.

My understanding here is that `message` is sent as an object (because we want to consume meta data like id and annotations) from `AssistantResponse` so it uses the `data_message` protocol. 

Also since `data_message` expects a [specific shape](https://github.com/vercel/ai/blob/4f28469a1af9f805a682ceb4bbd2d999a3b9add8/packages/ui-utils/src/stream-parts.ts#L165), I created an `assistant_message` stream part that exclusively handles the assistant APIs events so we can extend its functionality as the API changes within this stream part instead of creating new stream parts for every different event (`thread.create`, `thread.message.create`, etc.)